asm/load_store.go

```diff
@@ -111,7 +113,7 @@ func LoadMapPtr(dst Register, fd int) Instruction {
 		OpCode:   LoadImmOp(DWord),
 		Dst:      dst,
 		Src:      PseudoMapFD,
-		Constant: int64(fd),
+		Constant: int64(uint32(fd) & math.MaxUint32),
```

Comment:
Just for my reference: is the mask required here? Won't the u32 conversion take care of truncating already?

Reply:
You're right, it's not necessary.